### PR TITLE
warm-up assignment haha

### PR DESCRIPTION
**Pull request by MoranSin**

### CODE REVIEW

File: assignments/warm-up/MoranSin/routers/router.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+async def get_news(date: Optional[str] = None, time: Optional[str] = None):
+    if not validate_input(date, date_format) or not validate_input(time, time_format):
+        raise HTTPException(status_code=400, detail="Invalid input")
+    
+    data = get_all_news()
+    formatted_data = {}
+    # print("date or time", date , time)
+    if date:
+        day_data = data[date]
+        if date and time:
```
Comment (ShakedZrihen) [inline assignments/warm-up/MoranSin/routers/router.py:30]: date is truthy (you are inside it's `if`) so you can just ask for time

File: assignments/warm-up/MoranSin/routers/router.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            for news in day_data:
+                if time in news:
+                    formatted_data.update(news)
+        formatted_data = day_data
+    if time and not date:
+        for news_date in data:
+            for news in data[news_date]:
+                if time in news:
+                    formatted_data.update(news)
```
Comment (ShakedZrihen) [inline assignments/warm-up/MoranSin/routers/router.py:39]: im sure it can be done in a prettier way

File: assignments/warm-up/shaked/app.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ def breaking_news(date: Optional[str] = None, time: Optional[str] = None):
     for date in breaking_news_data:
         news_from_date = []
         for news in breaking_news_data[date]:
-            news_from_date.append({news: breaking_news_data[date][news]})
+            news_from_date.append({news: breaking_news_data[date][time]})
```
Comment (ShakedZrihen) [inline assignments/warm-up/shaked/app.py:21]: you don't need to update my code ;)

File: assignments/warm-up/MoranSin/tests/breaking_news_test_unit.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import requests
```
Comment (ShakedZrihen) [inline assignments/warm-up/MoranSin/tests/breaking_news_test_unit.py:1]: nice!

**Review by ShakedZrihen — CHANGES_REQUESTED** [review]
you are missing the README.md file, please add it as it a mandatory requirement

File: assignments/warm-up/MoranSin/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+fastapi==0.89.1
+mangum==0.17.0
+requests
+uvicorn
+BeautifulSoup4
+datetime
+dateutil.parser
```
Comment (ShakedZrihen) [inline assignments/warm-up/MoranSin/requirements.txt:7]: are you sure this lib exists?

**Comment by ShakedZrihen** [comment]
for this request for example:
`http://localhost:3000/breaking-news?date=28/07/2024&time=18:02`

this is what im getting back:
```
[
    {
        "18:02": " בן גביר מכחיש את הדיווח ב\"טלגרף\": \"מסרב לשתף פעולה עם תעמולה\""
    },
    {
        "17:58": " הותר לפרסום: קצין ולוחם בשריון נפצעו קשה בדרום רצועת עזה"
    },
    {
        "17:56": " הג'ודוקא גפן פרימו הפסידה בבית הניחומים, ולא תזכה בארד באולימפיאדה"
    },
    {
        "17:49": " צה\"ל הורה על פינוי אזורים במחנה אל-בורייג' בעזה: \"בעקבות ירי רקטות\""
    },
    {
        "17:38": " דיווח: \"חמאס תיעד עינוי חטופים ודרש מבן גביר הקלה בתנאי אסירים - הוא סירב\""
    },
    {
        "17:29": " חברת התעופה הלבנונית דחתה טיסות חוזרות - מבלי לציין את הסיבה"
    },
    {
        "17:19": " רוכב אופנוע נפצע קשה בתאונת דרכים עם רכב בכביש 4"
    },
    {
        "17:16": " המשטרה פיזרה את המפגינים בירושלים, 3 נעצרו"
    },
    {
        "17:11": " גבר עם סימני מכת חום אותר ברכב בערערה בנגב, מותו נקבע במקום"
    },
    {
        "17:06": " שריפה פרצה ליד כרמיאל, קו הבתים הראשון בשכונת גבעת רם מפונה"
    },
    {
        "17:03": " כ-20 פעילים חוסמים את הכניסה למשרדי הממשלה: \"עסקה עכשיו\""
    },
    {
        "16:49": " דיווח: לקראת תגובה ישראלית - חיזבאללה פינה מוצבים גם בסוריה"
    },
    {
        "16:39": " הפגיעה בבית בשתולה: חיזבאללה לקח אחריות על הירי"
    },
    {
        "16:30": " בריטניה מגנה את הטבח במג'דל שמס: \"חשש עמוק מהסלמה\""
    },
    {
        "16:19": " פגיעה ישירה בשתולה, אין נפגעים"
    },
    {
        "16:16": " בן 15 נעצר בצרפת בחשד שאיים ברצח על הנשיא הרצוג"
    },
    {
        "16:08": " דיווח בלבנון על תקיפה בדרום המדינה"
    },
    {
        "16:06": " דיווח: \"דיונים במטרה למנוע מישראל לתקוף ערים גדולות בלבנון - בהן ביירות\""
    },
    {
        "16:01": " אזעקה בגליל המערבי"
    },
    {
        "15:59": " לראשונה מאז הכחשת הירי למג'דל שמס: חיזבאללה לקח אחריות על ירי לישראל"
    },
    {
        "15:55": " נעצר אדם שחדר לבסיס צבאי באזור בית שמש - וחשוד כי חיבל בכלי רכב וגנב נשק"
    },
    {
        "15:49": " לפני הקבינט: נתניהו יכנס התייעצות עם בכירי מערכת הביטחון"
    },
    {
        "15:46": " קמפיין הבחירות של קמלה האריס: גייסנו תוך שבוע 200 מיליון דולר"
    },
    {
        "15:19": " השר בן צור: זה יום אבל לאומי, 12 ילדים נרצחו בטבח. על מדינות העולם לעמוד לצידנו"
    },
    {
        "15:16": " עקב הטבח: לשכת סוכני הביטוח הקימה מוקד חירום עבור תושבי מג'דל שמס"
    },
    {
        "14:34": " דיווח בלבנון על תקיפות בדרום המדינה"
    },
    {
        "14:11": " הרמטכ\"ל מאשר תוכניות לצפון, גלנט: \"חיזבאללה ירה - ויישא במחיר\""
    },
    {
        "13:59": " גפן פרימו הפסידה ברבע הגמר בעקבות איפון, תקווה לזכות במדליה דרך בית הניחומים"
    },
    {
        "13:35": " נתניהו נחת בנתב\"ג, צפוי לכנס בהקדם את הקבינט המדיני-ביטחוני"
    },
    {
        "13:13": " הג'ודוקא ברוך שמאילוב הפסיד בשמינית הגמר – והודח. גם הפעם יריבו לא לחץ את ידו"
    },
    {
        "13:07": " דיווח: חיזבאללה בכוננות גבוהה, פינה מתקנים מרכזיים לקראת \"הסלמה ישראלית\""
    },
    {
        "13:04": " אולימפיאדת פריז: גפן פרימו עלתה לרבע הגמר בג'ודו"
    },
    {
        "12:59": " בלינקן מאשר: \"הטיל שפגע בגולן שוגר על ידי חיזבאללה\""
    },
    {
        "12:50": " צבע אדום בעוטף"
    },
    {
        "12:48": " גורבנקו סיימה במקום השביעי במוקדמות ה-100 מטר חזה - ועלתה לחצי הגמר"
    },
    {
        "12:36": " לוקטב העפיל לחצי הגמר במשחה ל-200 מטר חזה"
    },
    {
        "12:29": " לבנון: \"ארה\"ב ביקשה שנרסן את חיזבאללה, תקיפה ישראלית משמעותית תוביל למלחמה אזורית\""
    },
    {
        "12:20": " ‏יו\"ר פורום הרשויות הדרוזיות לשרים: \"אל תהפכו את אירוע הטבח לפוליטי, אל תגיעו\""
    },
    {
        "12:01": " תושבים זועמים על שרי הקואליציה בהלוויות הנרצחים במג'דל שמס: \"עופו מפה!\""
    },
    {
        "11:58": " מליאת הכנסת אישרה סופית: התגמולים לחטופים ולנעדרים יעוגנו בחוק"
    },
    {
        "11:48": " ניצחונות לישראלים בג'ודו, היריב המרוקאי לא לחץ את ידו של שמאילוב"
    },
    {
        "11:42": " דיווח בלבנון: ירי ארטילרי בדרום המדינה"
    },
    {
        "11:33": " עקב הירי למג'דל שמס: מד\"א הודיעו על התרמת דם בדלית אל-כרמל, החל מ-15:00"
    },
    {
        "11:23": " החלה פסגת המו\"מ לעסקת החטופים ברומא"
    },
    {
        "10:43": " אישום: תושב רחובות ביצע עבירות מרמה בסך יותר מ-6 מיליון ש\"ח ע\"י זיוף צוואות"
    },
    {
        "10:25": " אלפים במג'דל שמס לקראת הלוויות המוניות, קרובי משפחה התעלפו מהדוחק ומצער"
    },
    {
        "10:20": " ברקע האסון בצפון: הבורסה נפתחה בירידות חדות"
    },
    {
        "10:07": " הטבח במג'דל שמס: חיפושים אחר אחד הילדים ששיחקו במגרש ונעדר מאז פגיעת הרקטה"
    },
    {
        "09:55": " מנהיג העדה הדרוזית: \"פגיעה באזרחים היא קו שחור, למה נתנו לחיזבאללה להתעצם ככה?\""
    },
    {
        "09:48": " משרד החוץ: זו ההזדמנות האחרונה לפתרון דיפלומטי בלבנון, לפני מלחמה הרסנית"
    },
    {
        "09:22": " דיווח בלבנון: כטב\"ם ישראלי תקף בכפר כילא שבדרום המדינה"
    },
    {
        "09:20": " ראש המוסד הגיע לרומא לפסגת המו\"מ לשחרור החטופים"
    },
    {
        "09:16": " איראן: \"מזהירים את ישראל מהרפתקה חדשה בלבנון, התגובה תהיה בלתי-צפויה\""
    },
    {
        "08:54": " טראמפ גינה את הטבח במג'דל שמס: \"חיזבאללה האכזרי פגע בילדים עם טיל איראני מדויק\""
    },
    {
        "08:26": " גלנט ביקר במג'דל שמס: \"טרגדיה נוראית, חיזבאללה אחראי - וישלם את המחיר\""
    },
    {
        "07:59": " הותרו לפרסום שמותיהם של 11 מהנערים והילדים שנרצחו בפגיעת הרקטה במג'דל שמס"
    },
    {
        "06:54": " צה\"ל מאשר: תקפנו בעומק ובדרום לבנון במהלך הלילה"
    },
    {
        "06:31": " תחזית מזג האוויר: עלייה בטמפרטורות, לצד הכבדה בעומסי החום"
    },
    {
        "05:59": " מת מפצעיו סמל יונתן אהרן גרינבלט, לוחם בגבעתי, שנפגע מנ\"ט ברפיח"
    },
    {
        "05:14": " שר החוץ של האיחוד האירופי גינה את הטבח במג'דל שמס: \"תמונות מזעזעות\""
    },
    {
        "04:21": " דיווחים ערביים על תקיפות במזרח לבנון, כ-90 ק\"מ צפונית למטולה"
    },
    {
        "03:03": " נתניהו המריא בדרכו חזרה לישראל"
    }
]
```

im expecting to get only the relevant time

**Comment by MoranSin** [comment]
oh my in the end I did forget the readme oops, but thanks for the critics ill try to fix them until tomorrow

**Pull request closed without merging**